tests/Benchmarks/System.Text.Primitives/PrimitiveParserBoolPerfTests.cs

```diff
@@ -0,0 +1,29 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Buffers.Text;
+using System.Collections.Generic;
+using BenchmarkDotNet.Attributes;
+using Benchmarks.System.Text.Primitives;
+
+namespace System.Text.Primitives.Benchmarks
+{
+    public partial class PrimitiveParserPerfTests
+    {
+        [Benchmark]
+        [Arguments("True")]
+        [Arguments("False")]
+        public void BaselineStringToBool(string text) => bool.TryParse(text, out _);
+
+        [Benchmark]
+        [ArgumentsSource(nameof(Utf8ByteArrays))]
+        public void PrimitiveParserByteSpanToBool(Utf8ByteArrayArgument text) => Utf8Parser.TryParse(text.CreateSpan(), out bool value, out int bytesConsumed);
```

Comment:
Anyway to get rid of the overhead for creating the span? Does it have much affect on the end results of these benchmarks, given parsing primitives like bool/ints is quire fast?

Reply:
@ahsonkhan this is a very good question.

I believe that this particular parsing operation should include the cost of the Span creation. Why? Because even if we have some huge buffer of memory and we want to parse just a part of it using the Span API we need to create a Span. So we can't use it without creating a Span instance.

If we would like to amortize the cost of Span creation we could use the `OperationsPerInvoke` feature:

```cs
[Benchmark(OperationsPerInvoke = 10)]
[ArgumentsSource(nameof(Utf8ByteArrays))]
public void PrimitiveParserByteSpanToBool(Utf8ByteArrayArgument text)
{
     bool result = default;
     var span = text.CreateSpan();

     result = Utf8Parser.TryParse(span, out bool value, out int bytesConsumed); // 1
     result = Utf8Parser.TryParse(span, out bool value, out int bytesConsumed); // 2
     result = Utf8Parser.TryParse(span, out bool value, out int bytesConsumed); // 3
     result = Utf8Parser.TryParse(span, out bool value, out int bytesConsumed); // 4
     result = Utf8Parser.TryParse(span, out bool value, out int bytesConsumed); // 5
     result = Utf8Parser.TryParse(span, out bool value, out int bytesConsumed); // 6
     result = Utf8Parser.TryParse(span, out bool value, out int bytesConsumed); // 7
     result = Utf8Parser.TryParse(span, out bool value, out int bytesConsumed); // 8
     result = Utf8Parser.TryParse(span, out bool value, out int bytesConsumed); // 9 
     result = Utf8Parser.TryParse(span, out bool value, out int bytesConsumed); // 10

    return result;
}
```